dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -159,11 +165,17 @@ bool TypeLookupService::extensibility(TypeFlag extensibility_mask, const TypeIde
   get_type_dependencies_i(type_ids, dependencies);
 
   for (unsigned i = 0; i < dependencies.length(); ++i) {
-    TypeObject dep_to = get_type_objects_i(dependencies[i].type_id);
-    if (TK_UNION == dep_to.minimal.kind) {
-      result = dep_to.minimal.union_type.union_flags & extensibility_mask;
-    } else if (TK_STRUCTURE == dep_to.minimal.kind) {
-      result = dep_to.minimal.struct_type.struct_flags & extensibility_mask;
+    const TypeObject& dep_to = get_type_objects_i(dependencies[i].type_id);
+    tk = dep_to.kind == EK_MINIMAL ? dep_to.minimal.kind : dep_to.complete.kind;
```

Comment:
```suggestion
    const TypeKind tk = dep_to.kind == EK_MINIMAL ? dep_to.minimal.kind : dep_to.complete.kind;
```

Reply:
I don't think defining a different constant variable `tk` here adds any benefit. Why not just use the same `tk` variable declared before?

Reply:
Because `const` data is easier to reason about, both for humans and compilers.